src/tools/ScenarioMeasurement/Startup/PerfCollect.cs

```diff
@@ -0,0 +1,136 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics.Tracing;
+using System.IO;
+using System.Reflection;
+using System.Text;
+
+namespace ScenarioMeasurement
+{
+    public class PerfCollect : IDisposable
+    {
+        private readonly string startupDirectory = Path.GetDirectoryName(Assembly.GetExecutingAssembly().Location);
+        private ProcessHelper perfCollectProcess;
+        public string TraceName { get; private set; }
+        public string TraceDirectory { get; private set; }
+        public string TraceFilePath { get; private set; }
+        private List<KernelKeyword> KernelEvents = new List<KernelKeyword>();
+        private List<ClrKeyword> ClrEvents = new List<ClrKeyword>();
+        public PerfCollect(string traceName, Logger logger) : this(traceName, Environment.CurrentDirectory, logger)
+        {
+        }
+
+        public PerfCollect(string traceName, string traceDirectory, Logger logger)
+        {
+            TraceName = traceName;
+            string perfCollectScript = Path.Combine(startupDirectory, "perfcollect");
+            if (!File.Exists(perfCollectScript))
+            {
+                throw new FileNotFoundException($"Pefcollect not found at {perfCollectScript}. Please rebuild the project to download it.");
+            }
+
+            if (String.IsNullOrEmpty(traceName))
+            {
+                throw new ArgumentException("Trace file name cannot be empty.");
+            }
+
+
+            if (!Directory.Exists(traceDirectory))
+            {
+                Directory.CreateDirectory(traceDirectory);
+            }
+            TraceDirectory = traceDirectory;
+
+
+            perfCollectProcess = new ProcessHelper(logger)
+            {
+                ProcessWillExit = true,
+                Executable = perfCollectScript,
+                Timeout = 300
+            };
+        }
+
+        public ProcessHelper.Result Start()
+        {
+            var arguments = new StringBuilder($"start {TraceName} -events ");
+
+            foreach (var keyword in KernelEvents)
+            {
+                arguments.Append(keyword.ToString());
+                arguments.Append(",");
+            }
+
+            foreach (var keyword in ClrEvents)
+            {
+                arguments.Append(keyword.ToString());
+                arguments.Append(",");
+            }
+
+            string args = arguments.ToString().TrimEnd(',');
```

Comment:
just FYI: Since you went to the trouble to use `StringBuilder`, the final `.TrimEnd()` here will end up allocating an extra string you don't need. Instead I would so something like `.Remove(arguments.Length-1,1)`  before you call `ToString()`. When you're using a `StringBuilder` you should do all your editing and manipulation on the `StringBuilder` instance and then think of calling `ToString()` as "finalizing" the string.